frontend/src/app/main/data/workspace/libraries_helpers.cljs

```diff
@@ -532,13 +532,26 @@
   (log/debug :msg "Sync shape direct" :shape (str shape-id) :reset? reset?)
   (let [shape-inst     (ctn/get-shape container shape-id)]
     (if (ctk/in-component-copy? shape-inst)
-      (let [library        (dm/get-in libraries [(:component-file shape-inst) :data])
-            component      (or (ctkl/get-component library (:component-id shape-inst))
-                               (and reset?
-                                    (ctkl/get-deleted-component library (:component-id shape-inst))))
-
-            shape-main     (when component
-                             (ctf/get-ref-shape library component shape-inst))
+      (let [redirect-shaperef ;;Set the :shape-ref of a shape pointing to the :id of its remote-shape
```

Comment:
I think this function doesn't add much value. Perhaps the code will be clearer if you just put the contents inline in the two places they are used. What do you think?